File: package/main/src/String/titleCase.ts
import { capitalizeWord } from "./capitalizeWord";
import { words } from "./words";

/**
 * Converts a string to Title Case (each word capitalized, separated by spaces).
 * @param {string} string_ - Input string
 * @returns {string} Title Case string
 * @example
 * titleCase("hello world"); // "Hello World"
 * titleCase("a-quick brown_fox"); // "A Quick Brown Fox"
 */
export const titleCase = (string_: string): string =>
  words(string_)
    .map((word) => capitalizeWord(word))
    .join(" ");

File: package/umt_python/src/string/title_case.py
from .capitalize_word import capitalize_word
from .words import words


def title_case(string_: str) -> str:
    """
    Converts a string to Title Case (each word capitalized, separated by
    spaces).

    Args:
        string_: Input string.

    Returns:
        Title Case string.

    Example:
        >>> title_case("hello world")
        'Hello World'
        >>> title_case("a-quick brown_fox")
        'A Quick Brown Fox'
        >>> title_case("helloWorld")
        'Hello World'
    """
    return " ".join(capitalize_word(word) for word in words(string_))
